iceoryx_hoofs/source/posix_wrapper/shared_memory_object/shared_memory.cpp

```diff
@@ -68,6 +68,16 @@ expected<SharedMemory, SharedMemoryError> SharedMemoryBuilder::create() noexcept
 
     auto nameWithLeadingSlash = addLeadingSlash(m_name);
 
+    bool hasOwnership = (m_openMode == OpenMode::EXCLUSIVE_CREATE || m_openMode == OpenMode::PURGE_AND_CREATE
+                         || m_openMode == OpenMode::OPEN_OR_CREATE);
+
+    if (hasOwnership && (m_accessMode == AccessMode::READ_ONLY))
+    {
+        std::cerr << "Cannot create shared-memory file \"" << m_name << "\" in read-only mode. "
+                  << "Initializing a new file requires write access" << std::endl;
+        return cxx::error<SharedMemoryError>(SharedMemoryError::INCOMPATIBLE_OPEN_AND_ACCESS_MODE);
```

Comment:
@tobiasblass we moved some code around to create a flat include structure and namespace. `iox::cxx::error` is now just `iox::error`. Removing the `cxx::` should fix the CI